### PR TITLE
Add "Kill Scuttlebugs" objective

### PR DESCRIPTION
**Pull request by zcoop98**
Adds the "Kill Scuttlebugs" objective to the set of possible bingo objectives.

---

The `MAX_SCUTTLEBUGS` value in `bingo_tracking_collectibles.c` is set at 15.

The quantity range for the objective is set to 4-9 enemies; there are 9 unique Scuttlebugs in the game (3 in BBH, 6 in HMC).
This objective is weighted as a medium-difficulty objective on the bingo board (like the other "kill [enemy]" objectives).

### CODE REVIEW

**Comment by zcoop98** [comment]
What do you think about the 4-9 range for the objective? 

There are 3 Scuttlebugs outside the mansion in BBH, the other 6 are in HMC.
The first 5 HMC Scuttlebugs are pretty trivial to find, the last is all the way down in the "Metal-Head Mario can move!" star corridor.

File: src/game/behaviors/scuttlebug.inc.c [hunk 1]
```diff
@@ -27,14 +27,18 @@ void bhv_scuttlebug_loop(void) {
...
+        && obj_set_hitbox_and_die_if_attacked_bingo(&sScuttlebugHitbox, SOUND_OBJ_DYING_ENEMY1,
+                                              o->oScuttlebugUnkF4, BINGO_UPDATE_KILLED_SCUTTLEBUG))
+        o->oSubAction = 4;
+    if (o->oSubAction != 2)
         o->oScuttlebugUnkF8 = 0;
     switch (o->oSubAction) {
         case 0:
+            o->oBingoId = get_unique_id(BINGO_UPDATE_KILLED_SCUTTLEBUG, o->oPosX, o->oPosY, o->oPosZ);
+            o->oSubAction = 1;
+            break;
```
Comment (matt-kempster) [inline src/game/behaviors/scuttlebug.inc.c:40]: I think this technically changes the behavior of Scuttlebugs by adding an extra frame before they are initialized. I generally try to keep game behavior as identical as possible to vanilla SM64. Instead, could we just add this logic into the beginning of the original `case 0`?

**Review by matt-kempster — CHANGES_REQUESTED** [review]
I think the 4-9 range is totally reasonable for now, though I do agree that eventually separating it into easy and medium would be great. I think the whole difficulty subsystem needs a revamp anyway (my main issue with it is that some things are easy but take a long time, and some things are hard but could potentially take very little time) so I wouldn't think about it too hard.

Requesting changes for the Scuttlebug change, otherwise this is good to go!

File: src/game/behaviors/scuttlebug.inc.c (hunk 1 above)
Comment (zcoop98) [reply src/game/behaviors/scuttlebug.inc.c]: How's this? 

https://github.com/zcoop98/bingo64/blob/591ca626464eb5acd242187f83596567b302f090/src/game/behaviors/scuttlebug.inc.c#L36-L41

I still need a way to set the `BingoId` on first loop and never again, so I dropped the `break`.
This should mean that execution now goes straight into the original step 0, but still never hits the _new_ step 0 again.

Comment (zcoop98) [reply src/game/behaviors/scuttlebug.inc.c]: For reference, I got the idea from the [spindrift bingo code](https://github.com/matt-kempster/bingo64/blob/master/src/game/behaviors/spindrift.inc.c#L24-L29), which uses the same mechanism:

```c
switch (o->oAction) {
    case 0:
        o->oBingoId = get_unique_id(BINGO_UPDATE_KILLED_SPINDRIFT, o->oPosX, o->oPosY, o->oPosZ);
        o->oAction = 1;
        break;
    case 1:
```

Comment (matt-kempster) [reply src/game/behaviors/scuttlebug.inc.c]: Ah wow you are TOTALLY right!

Yes, I very much approve of the `// fallthrough` comment, and am sad that I broke my own recommendation about the Spindrifts! Definitely think I should go back and fix that.

**Review by matt-kempster — APPROVED** [review]
Thank you very much for adding this! I look forward to any more contributions you care to make!

**Pull request merged**